sdk/cosmos/src/errors.rs

```diff
@@ -9,17 +9,36 @@ pub enum Error {
     /// An error related to parsing
     #[error(transparent)]
     ParsingError(#[from] ParsingError),
-    #[error("http error: {0}")]
-    CoreHttpError(#[from] azure_core::HttpError),
-    #[error("stream error: {0}")]
-    StreamError(#[from] azure_core::StreamError),
-    #[error("http error: {0}")]
-    HttpError(#[from] http::Error),
-    #[error("JSON error: {0}")]
-    JsonError(#[from] serde_json::Error),
-    /// Other errors that can happen but are unlikely to be matched against
-    #[error(transparent)]
-    Other(#[from] Box<dyn std::error::Error + Send + Sync>),
+}
+
+impl From<serde_json::Error> for Error {
```

Comment:
Not: all these `From` impls could easily be performed by a macro defined in `core` as all SDK error types will want to do the same thing here.